ecosystem/typescript/sdk_v2/src/transactions/types/rawTransaction.ts

```diff
@@ -0,0 +1,198 @@
+ */
+export abstract class RawTransactionWithData {
+  /**
+   * Serialize a Raw Transaction With Data
+   */
+  abstract serialize(serializer: Serializer): void;
+
+  /**
+   * Deserialize a Raw Transaction With Data
+   */
+  static deserialize(deserializer: Deserializer): RawTransactionWithData {
+    const index = deserializer.deserializeUleb128AsU32();
+    /**
+     * index is represented in rust as an enum
+     * {@link https://github.com/aptos-labs/aptos-core/blob/main/types/src/transaction/mod.rs#L440}
+     */
+
+    switch (index) {
+      case 0:
+        return MultiAgentRawTransaction.load(deserializer);
+      case 1:
+        return FeePayerRawTransaction.load(deserializer);
+      default:
+        throw new Error(`Unknown variant index for RawTransactionWithData: ${index}`);
+    }
+  }
+}
+
+/**
+ * Representation of a Multi Agent Transaction that can serialized and deserialized
+ */
+export class MultiAgentRawTransaction extends RawTransactionWithData {
+  /**
+   * The raw transaction
+   */
+  public readonly raw_txn: RawTransaction;
+
+  /**
+   * The secondary signers on this transaction
+   */
+  public readonly secondary_signer_addresses: Array<AccountAddress>;
+
+  constructor(raw_txn: RawTransaction, secondary_signer_addresses: Array<AccountAddress>) {
+    super();
+    this.raw_txn = raw_txn;
+    this.secondary_signer_addresses = secondary_signer_addresses;
+  }
+
+  serialize(serializer: Serializer): void {
+    // enum variant index
+    serializer.serializeU32AsUleb128(0);
+    this.raw_txn.serialize(serializer);
+    serializer.serializeVector<TransactionArgument>(this.secondary_signer_addresses);
+  }
+
+  static load(deserializer: Deserializer): MultiAgentRawTransaction {
+    const rawTxn = RawTransaction.deserialize(deserializer);
+    const secondarySignerAddresses = deserializer.deserializeVector(AccountAddress);
+
+    return new MultiAgentRawTransaction(rawTxn, secondarySignerAddresses);
+  }
+}
+
+/**
+ * Representation of a Fee Payer Transaction that can serialized and deserialized
+ */
+export class FeePayerRawTransaction extends RawTransactionWithData {
+  /**
+   * The raw transaction
+   */
+  public readonly raw_txn: RawTransaction;
+
+  /**
+   * The secondary signers on this transaction - optional and can be empty
+   */
+  public readonly secondary_signer_addresses: Array<AccountAddress>;
+
+  /**
+   * The fee payer account address
+   */
+  public readonly fee_payer_address: AccountAddress;
+
+  constructor(
+    raw_txn: RawTransaction,
+    secondary_signer_addresses: Array<AccountAddress>,
+    fee_payer_address: AccountAddress,
+  ) {
+    super();
+    this.raw_txn = raw_txn;
+    this.secondary_signer_addresses = secondary_signer_addresses;
+    this.fee_payer_address = fee_payer_address;
+  }
+
+  serialize(serializer: Serializer): void {
+    // enum variant index
+    serializer.serializeU32AsUleb128(1);
```

Comment:
Note for these numbers, maybe we make an enum that matches rust for easy identification, and then just use it here for the number

Reply:
good idea!